### DISCUSSION TITLE
List of FMX features not handled

### DISCUSSION THREAD

**Discussion by srondeauuwt** [Embarcadero/DelphiFMX4Python#26] [Ideas]
It didn't take long to realize that MessageDialog and possibly its parent DialogService were not included in the delphifmx package. That's because I started to port my AutoitScript program to DelphiFMX4Python in the hopes that it would make it cross-platform instead of Windows-only -- we have a lot of students who have Macs here as well as Windows, and some that have Linux as their primary platform. 

Regardless, it would be good to know what is present in Delphi FMX that is NOT in DelphiFMX4Python, so we can reference something directly as we port. I did sort of grep on the functions imported from delphifmx, but that's not the best solution and does not show us what we don't look for. 

I had thought DelphiFMX4Python included all of FMX, not just a subset of functions. Comparing the python version of ControlsDemo to the Delphi version, I realized there were multiple tabs with advanced features in the Delphi version -- and only one "Standard" tab with basic GUI widgets in the Python version. I know you guys are busy and doing your best, but documentation stating that this is a subset or a work in process would help people understand its limitations.

**Comment by Priyatham10** [comment]
Thank you @srondeauuwt for the idea. Great to see you porting to DelphiFMX4Python. We're doing incremental work. We shall try to provide that information.

**Comment by lmbelo** [comment]
The DialogService is now available 💪 

![Screenshot 2023-03-07 at 17 52 29](https://user-images.githubusercontent.com/8376898/223549778-97aa0339-a5a0-48a8-9f1e-c6598caedfee.png)
